lib/pbench/server/api/resources/upload_api.py

```diff
@@ -91,11 +91,16 @@ def put(self, controller):
                 f"Tarfile upload: Invalid content-length header, not an integer for file {filename}"
             )
             abort(400, message="Invalid content-length header, not an integer")
-        except Exception:
-            self.logger.exception(
+        except KeyError:
+            self.logger.warning(
                 f"Tarfile upload: No Content-Length header value found for file {filename}"
             )
             abort(400, message="Missing required content-length header")
```

Comment:
Huh. I was thinking at first that if we're bothering with `Exception` here, why not in the other header checks? Then I realized that we're using `.get` so we'll just get `None` rather than a `KeyError`; I think that would just be another `ValueError`, and the only way we'll see "no Content-Length" would be if we checked for `None` explicitly.

I wonder if all of the headers should use the optimistic pattern of

```
try:
    h = request.headers["header"]
    g = request.headers["gopher"]
except KeyError as e:
    self.logger.warning("missing required header {}", e)
    abort(400, f"Missing required {e} header")
```

`Content-Length`, with the `int()` conversion, is a special case, adding the `ValueError`, so we might not want to just put them all in a list like this, but it _would_ be good to follow the same pattern

In any case, though; the original code here was wrong about the meaning of `Exception`, because anything other than `ValueError` is a 500 Internal Error; and if we want to catch "missing header" we need to do it like the others, by checking explicitly for `None` or by using array reference style and catching `KeyError`.